### ISSUE TITLE
Need some understanding about deploy hook

### ISSUE THREAD

**Issue by prahladyeri** [openshift/wordpress-example#86]
This is not a bug, I just have a query about [deploy](https://github.com/openshift/wordpress-example/blob/master/.openshift/action_hooks/deploy) hook and I did not know where to ask it. Since I'm soon going to create my own Openshift Quickstart app for a PHP framework, I wanted to know this:

1. My first query is why did you first copy the entire contents of `install_dir` to `dest_dir` when you could have done the same work from the `install_dir` itself?

     cp -rf ${install_dir}/* ${dest_dir}/

2. My second question is that you are creating too many symlinks. Initially, you link the repo's `php` directory to the `$OPENSHIFT_DATA_DIR/current`:

```
if [ -d ${OPENSHIFT_REPO_DIR}php ]; then
  rm -rf ${OPENSHIFT_REPO_DIR}php
fi
ln -sf ${dest_dir} ${OPENSHIFT_REPO_DIR}php

```
However, `$OPENSHIFT_DATA_DIR/current` soon gets empty as you copy the contents like plugins, themes, etc. to it's parent `$OPENSHIFT_DATA_DIR`. So, then again you have to create symlinks inside the php directory to point to each one of these like this:

```
ln -sf ${OPENSHIFT_DATA_DIR}blogs.dir ${OPENSHIFT_REPO_DIR}php/wp-content/
ln -sf ${OPENSHIFT_DATA_DIR}plugins   ${OPENSHIFT_REPO_DIR}php/wp-content/
ln -sf ${OPENSHIFT_DATA_DIR}themes    ${OPENSHIFT_REPO_DIR}php/wp-content/
ln -sf ${OPENSHIFT_DATA_DIR}languages    ${OPENSHIFT_REPO_DIR}php/wp-content/
ln -sf ${OPENSHIFT_DATA_DIR}uploads   ${OPENSHIFT_REPO_DIR}php/wp-content/

```
However, isn't it a much simpler strategy to just symlink the php directory to {$OPENSHIFT_DATA_DIR} so it automatically gets the themes, plugins, etc. from there?

Thanks.

**Comment by bparees** [comment]
@sallyom @dinhxuanvu can you help out here?  (v2)

**Comment by sallyom** [comment]
- For changes/updates to the WordPress app to be persistent, they must be made via git and copied to the persistent storage directory upon deployment.  This results in a current working version of your app where blogs/plugins/themes/languages/uploads can be updated/modified and then set in stone upon deployment (git commit/push).  The deploy process creates symlinks to map the default WordPress file locations to the new locations in $OPENSHIFT_DATA_DIR. 
 - Updates/modifications are deployed with a clean WordPress installation
 - Copying to dest_dir is necessary for scaling of WordPress applications.
In short, the awkward file structure is WordPress-specific.   Check out our other quickstarts  for more examples.  (https://github.com/openshift/cakephp-example)
